dev/download_tbb.sh

```diff
@@ -18,65 +19,96 @@
-TBB_URL="${TBB_URL_ROOT}/${TBB_PACKAGE}"
-DST=$(dirname "$0")/../__deps/tbb
-mkdir -p "${DST}/${OS}"
-DST=$(cd "${DST}" || exit 1;pwd)
+  TBB_PACKAGE="oneapi-tbb-${TBB_VERSION}-${TBB_OS}.${ARCH_EXT}"
+  TBB_URL="${TBB_URL_ROOT}/${TBB_PACKAGE}"
+  DST=$(dirname "$0")/../__deps/tbb
+  mkdir -p "${DST}/${OS}"
+  DST=$(cd "${DST}" || exit 1;pwd)
 
-DOWNLOAD_CODE=1
+  DOWNLOAD_CODE=1
 
-if [ ! -d "${DST}/${OS}/bin" ]; then
-  if [ -x "$(command -v curl)" ]; then
-    echo curl -L -o "${DST}/${TBB_PACKAGE}" "${TBB_URL}"
-    if curl -L -o "${DST}/${TBB_PACKAGE}" "${TBB_URL}";
-    then
-      DOWNLOAD_CODE=0
+  if [ ! -d "${DST}/${OS}/bin" ]; then
+    if [ -x "$(command -v curl)" ]; then
+      echo curl -L -o "${DST}/${TBB_PACKAGE}" "${TBB_URL}"
+      if curl -L -o "${DST}/${TBB_PACKAGE}" "${TBB_URL}";
+      then
+        DOWNLOAD_CODE=0
+      fi
+    elif [ -x "$(command -v wget)" ]; then
+      echo wget -O "${DST}/${TBB_PACKAGE}" "${TBB_URL}"
+      if wget -O "${DST}/${TBB_PACKAGE}" "${TBB_URL}";
+      then
+        DOWNLOAD_CODE=0
+      fi
+    else
+      echo "curl or wget not available"
+      exit 1
     fi
-  elif [ -x "$(command -v wget)" ]; then
-    echo wget -O "${DST}/${TBB_PACKAGE}" "${TBB_URL}"
-    if wget -O "${DST}/${TBB_PACKAGE}" "${TBB_URL}";
-    then
-      DOWNLOAD_CODE=0
-    fi
-  else
-    echo "curl or wget not available"
-    exit 1
-  fi
 
-  if [ ${DOWNLOAD_CODE} -ne 0 ] || [ ! -e "${DST}/${TBB_PACKAGE}" ]; then
-    echo "Download from ${TBB_URL} to ${DST} failed"
-    exit 1
-  fi
+    if [ ${DOWNLOAD_CODE} -ne 0 ] || [ ! -e "${DST}/${TBB_PACKAGE}" ]; then
+      echo "Download from ${TBB_URL} to ${DST} failed"
+      exit 1
+    fi
 
-  if [ "${OS}" = "win" ]; then
-    echo unzip -d "${DST}/${OS}" "${DST}/${TBB_PACKAGE}"
-    unzip -d "${DST}/${OS}" "${DST}/${TBB_PACKAGE}"
-    mv "${DST}/${OS}/oneapi-tbb-${TBB_VERSION}" "${DST}/${OS}/tbb"
+    if [ "${OS}" = "win" ]; then
+      echo unzip -d "${DST}/${OS}" "${DST}/${TBB_PACKAGE}"
+      unzip -d "${DST}/${OS}" "${DST}/${TBB_PACKAGE}"
+      mv "${DST}/${OS}/oneapi-tbb-${TBB_VERSION}" "${DST}/${OS}/tbb"
+    else
+      echo tar -xvf "${DST}/${TBB_PACKAGE}" -C "${DST}"
+      tar -C "${DST}/${OS}" --strip-components=1 -xvf "${DST}/${TBB_PACKAGE}"
+    fi
+    ls -al "${DST}/${OS}/"
+    echo "Downloaded and unpacked oneTBB to ${DST}/${OS}"
   else
-    echo tar -xvf "${DST}/${TBB_PACKAGE}" -C "${DST}"
-    tar -C "${DST}/${OS}" --strip-components=1 -xvf "${DST}/${TBB_PACKAGE}"
+    echo "oneTBB is already installed in ${DST}/${OS}"
   fi
-  ls -al "${DST}/${OS}/"
-  echo "Downloaded and unpacked oneTBB to ${DST}/${OS}"
+
 else
-  echo "oneTBB is already installed in ${DST}/${OS}"
-fi
+  sudo apt-get update
+  sudo apt-get install build-essential gcc gfortran cmake -y
+  git clone https://github.com/oneapi-src/oneTBB.git
+  CoreCount=$(lscpu -p | grep -Ev '^#' | wc -l)
+
+  rm -rf __deps/tbb
+  pushd oneTBB
+    git checkout v2021.11.0
```

Comment:
even once this would be in separate script it would worth placing dependency version at least to the header of the file